core/src/test/java/org/apache/brooklyn/util/core/task/ValueResolverTest.java

```diff
@@ -69,46 +59,65 @@ public void testTimeoutBig() {
         Assert.assertEquals(result.get(), "foo");
     }
 
-    public void testNoExecutionContextOnCompleted() {
+    public void testCompletedTaskReturnsResultImmediately() {
+        // Call ValueResolver.getMaybe() from this thread, which has no execution context.
```

Comment:
I still struggled with understanding what was meant here until I realised that this line 63 refers to the `Tasks.resolving(t)` being executed in this thread (I'd been put off by concentrating on the line `app.getExecutionContext()...`).  It would maybe make this clearer if these lines of comment were moved to just above the `Tasks.resolving(t)` on 68 below.